libs/sdk-ui-ext/src/internal/dashboardEmbedding/DashboardView/DashboardView.tsx

```diff
@@ -0,0 +1,80 @@
+// (C) 2020 GoodData Corporation
+import React, { useEffect } from "react";
+import { ErrorComponent as DefaultError, LoadingComponent as DefaultLoading } from "@gooddata/sdk-ui";
+import { ThemeProvider, useThemeIsLoading } from "@gooddata/sdk-ui-theme-provider";
+import { useDashboard } from "../useDashboard";
+import { DashboardRenderer } from "./DashboardRenderer";
+import { useDashboardAlerts } from "../useDashboardAlerts";
+import { IDashboardViewProps } from "./types";
+
+export const DashboardView: React.FC<IDashboardViewProps> = ({
+    dashboard,
+    filters,
+    theme,
+    backend,
+    workspace,
+    onDrill,
+    drillableItems,
+    onError,
+    onDashboardLoaded,
+    ErrorComponent = DefaultError,
+    LoadingComponent = DefaultLoading,
+}) => {
+    const { error: dashboardError, result: dashboardData, status: dashboardStatus } = useDashboard({
+        dashboard,
+        onError,
+    });
+
+    const { error: alertsError, result: alertsData, status: alertsStatus } = useDashboardAlerts({
+        dashboard,
+        onError,
+    });
+
+    const isThemeLoading = useThemeIsLoading();
+    const hasThemeProvider = isThemeLoading !== undefined;
+
+    useEffect(() => {
+        if (alertsData && dashboardData) {
+            onDashboardLoaded?.({
+                alerts: alertsData,
+                dashboard: dashboardData,
+            });
+        }
+    }, [onDashboardLoaded, alertsData, dashboardData]);
+
+    const statuses = [dashboardStatus, alertsStatus];
+
+    if (statuses.includes("loading") || statuses.includes("pending")) {
+        return <LoadingComponent />;
+    }
+
+    if (dashboardStatus === "error") {
+        return <ErrorComponent message={dashboardError.message} />;
+    }
+
+    if (alertsStatus === "error") {
+        return <ErrorComponent message={alertsError.message} />;
+    }
+
+    const dashboardRender = (
+        <DashboardRenderer
+            dashboard={dashboardData}
+            alerts={alertsData}
+            filters={filters}
+            onDrill={onDrill}
+            drillableItems={drillableItems}
+            ErrorComponent={ErrorComponent}
+            LoadingComponent={LoadingComponent}
+        />
+    );
+
+    if (!hasThemeProvider) {
```

Comment:
Should we do this @lupko? I spoke to @ivan-nejezchleb about this and since the Themes are not scopable (thanks IE11), we should probably tell the user to create a theme provider themselves to avoid surprise theming outside of the DashboardView.

Reply:
how about a prop to influence this. document the limitation of the theme provider so that people understand. keep the theme provider in as default, allow turning it off.

Reply:
to fill in extra thoughts.. i think that approach is more forward compatible with ideal state. in ideal state, we would have styled components & the styling would apply to dashboard only. also, the style loading may be more intricate (load style for dashboard, not from project). it makes more sense (i think) to keep the theming encapsulated within the dashboard view.

only as a workaround for the mess in the current status quo, we keep the flag to turn on the styling behavior.

Reply:
makes sense, thanks, I will incorparate this as soon as possible (probably tomorrow, though)

Reply:
updated